src/foraging_gui/Dialogs.py

```diff
@@ -2335,7 +2335,7 @@ def __init__(self, MainWindow, parent=None):
         self._setup_allbacks()
         
         # Sync selected subject_id
-        self.update_auto_train_fields(subject_id=self.MainWindow.ID.text())
+        self.update_auto_train_fields(subject_id=self.MainWindow.behavior_session_model.subject)
```

Comment:
I think this is the only line related to autotrain, and it has been taken care of.